### PR TITLE
Remove legacy modules

### PR DESCRIPTION
**Pull request by paulbakker**
Preview branch removing all legacy modules. This will be merged in late 2024. From then on, the DGS/Spring GraphQL integration will be the only implementation.

### Test with the Release Candidate
We published `10.0.0-rc.1` as the first Release Candidate based on this branch.
To test with the RC, add the following repository to your `build.gradle` (or Maven equivalent). 

```
repositories {
 maven {
        name = "OSS Candidates"
        url = "https://artifacts-oss.netflix.net/artifactory/maven-oss-candidates"
    }
}
```

**### Details of changes**
The following modules are completely removed because they are replaced by Spring GraphQL. If you were already using the `com.netflix.graphql.dgs:graphql-dgs-spring-graphql-starter` (Spring GraphQL integration) you were already not using any of these modules.

**Deleted modules:** 
- graphql-dgs-spring-boot-oss-autoconfigure
- graphql-dgs-spring-webmvc
- graphql-dgs-spring-webmvc-autoconfigure
- graphql-dgs-spring-boot-starter
- graphql-dgs-example-java
- graphql-dgs-example-java-webflux
- graphql-dgs-subscriptions-websockets
- graphql-dgs-subscriptions-websockets-autoconfigure
- graphql-dgs-subscriptions-graphql-sse
- graphql-dgs-subscriptions-graphql-sse-autoconfigure
- graphql-dgs-subscriptions-sse
- graphql-dgs-subscriptions-sse-autoconfigure
- graphql-dgs-spring-webflux-autoconfigure
- graphql-dgs-webflux-starter

**Deleted classes:**
- **DgsAutoConfiguration**: Autoconfiguration classes have moved. This may break tests that are using `@SpringBootTest(classes = {DgsAutoConfiguration.class, ...}`, and should use `@EnableDgsTest` instead.
- **DefaultGraphQLClient**: This is a long deprecated class that has been replaced by `[CustomGraphQLClient`, `CustomReactiveGraphQLClient` and `WebClientGraphQLClient`.
- **DefaultDgsReactiveQueryExecutor**: There should be no user impact because its interface should be used instead.
- **DefaultDgsQueryExecutor**: There should be no user impact because its interface should be used instead.

**Other changes**
- Subscriptions over SSE in Spring GraphQL are using the newer, more official [GraphQL over SSE RFC spec](https://github.com/graphql/graphql-over-http/blob/main/rfcs/GraphQLOverSSE.md). The old spec is no longer supported by DGS.
- If you are using `SSESubscriptionGraphQLClient` in tests to test your DGS server, switch to `GraphqlSSESubscriptionGraphQLClient`, which has the same interface, but uses the new protocol.
- `SSESubscriptionGraphQLClient` can still be used to call into other services using the old protocol, but this client is now deprecated for removal in the future.
- Note that Spring GraphQL serves subscriptions on the `/graphql` endpoint, not on the `/subscriptions` endpoint like DGS used to do.

### CODE REVIEW

**Comment by paulbakker** [comment]
@srinivasankavitha @kilink Preparing this branch for onboarding purposes. Not ready to merge yet, but it's a preview of things to come.
Note that I've disabled some client SSE/WS tests for now, need to replace some things in the setup with Spring GraphQL to get those to work.

**Comment by srinivasankavitha** [comment]
Looks good to me. I've noticed quite a few projects that add `graphql-dgs-spring-boot-starter` as a dependency in their `build.gradle` files in addition to the starter. This is incorrect usage in any case. Calling it out explicitly so we are not surprised by breakages there.

File: graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/RestClientGraphQLClientTest.kt [hunk 1]
```diff
@@ -50,17 +47,15 @@ import org.springframework.web.client.HttpClientErrorException
 import org.springframework.web.client.RestClient
 import java.time.LocalDateTime
 import java.time.format.DateTimeFormatter
-import java.util.Locale
+import java.util.*
```
Comment (iuliiasobolevska) [inline graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/RestClientGraphQLClientTest.kt:50]: extra nit: revert back to explicit imports

File: graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/GraphqlSSESubscriptionGraphQLClientTest.kt [hunk 2]
```diff
@@ -16,33 +16,36 @@
...
-@Disabled("Suspecting this test class to be the root cause of hanging builds")
+@EnableDgsTest
 internal class GraphqlSSESubscriptionGraphQLClientTest {
     @LocalServerPort
     var port: Int? = null
 
     @Test
     fun `A successful subscription should publish ticks`() {
-        val client = GraphqlSSESubscriptionGraphQLClient("/subscriptions", WebClient.create("http://localhost:$port"))
+        val client = GraphqlSSESubscriptionGraphQLClient("/graphql", WebClient.create("http://localhost:$port"))
```
Comment (iuliiasobolevska) [inline graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/GraphqlSSESubscriptionGraphQLClientTest.kt:48]: TODO: document in the PR description `/subscriptions` changes

File: graphql-dgs-example-shared/build.gradle [hunk 3]
```diff
@@ -15,12 +15,14 @@
...
     implementation(project(":graphql-dgs-pagination"))
     implementation(project(":graphql-dgs-extended-scalars"))
     implementation("io.projectreactor:reactor-core")
     implementation("org.springframework:spring-context")
     implementation("org.springframework:spring-web")
     implementation("com.fasterxml.jackson.core:jackson-databind")
     testImplementation ('io.projectreactor:reactor-test')
+    testImplementation(project(":graphql-dgs-spring-graphql-starter-test"))
+    testImplementation(project(":graphql-dgs-spring-graphql-starter"))
+    testImplementation("name.nkonev.multipart-spring-graphql:multipart-spring-graphql:1.+")
```
Comment (iuliiasobolevska) [inline graphql-dgs-example-shared/build.gradle:27]: is `name.nkonev.multipart-spring-graphql:multipart-spring-graphql:1.+` really in use?

File: graphql-dgs-extended-scalars/build.gradle.kts [hunk 4]
```diff
@@ -19,5 +19,5 @@ dependencies {
     api("com.graphql-java:graphql-java-extended-scalars")
     implementation("org.springframework.boot:spring-boot-autoconfigure")
 
-    testImplementation(project(":graphql-dgs-spring-boot-starter"))
+    testImplementation(project(":graphql-dgs-spring-graphql-starter"))
```
Comment (iuliiasobolevska) [inline graphql-dgs-extended-scalars/build.gradle.kts:22]: nit: weird to have two `graphql`s in the name but I guess it is what it is

File: graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/GraphqlSSESubscriptionGraphQLClientTest.kt [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2023 Netflix, Inc.
+ * Copyright 2021 Netflix, Inc.
```
Comment (iparadiso) [inline graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/GraphqlSSESubscriptionGraphQLClientTest.kt:2]: More curious than an actual comment, but why did the (c) get adjusted back in time?  Should this be a SED to bump them up to 2024?

File: graphql-dgs-spring-graphql/src/main/kotlin/com/netflix/graphql/dgs/springgraphql/autoconfig/DgsSpringGraphQLAutoConfiguration.kt [hunk 6]
```diff
@@ -68,17 +119,321 @@ import org.springframework.web.reactive.result.method.annotation.CookieValueMeth
 import org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter
 import org.springframework.web.servlet.mvc.method.annotation.ServletCookieValueMethodArgumentResolver
 import org.springframework.web.servlet.mvc.method.annotation.ServletRequestDataBinderFactory
-import java.util.Optional
+import java.time.Duration
+import java.util.*
```
Comment (iuliiasobolevska) [inline graphql-dgs-spring-graphql/src/main/kotlin/com/netflix/graphql/dgs/springgraphql/autoconfig/DgsSpringGraphQLAutoConfiguration.kt:123]: extra nit: use explicit imports

File: graphql-dgs-spring-boot-oss-autoconfigure/src/main/kotlin/com/netflix/graphql/dgs/autoconfig/DgsAutoConfiguration.kt [hunk 7]
```diff
@@ -29,7 +29,6 @@ import com.netflix.graphql.dgs.context.GraphQLContextContributor
 import com.netflix.graphql.dgs.context.GraphQLContextContributorInstrumentation
 import com.netflix.graphql.dgs.exceptions.DefaultDataFetcherExceptionHandler
 import com.netflix.graphql.dgs.internal.*
-import com.netflix.graphql.dgs.internal.DefaultDgsQueryExecutor.ReloadSchemaIndicator
```
Comment (iuliiasobolevska) [inline graphql-dgs-spring-boot-oss-autoconfigure/src/main/kotlin/com/netflix/graphql/dgs/autoconfig/DgsAutoConfiguration.kt:32]: file comment: is this supposed to be removed in favor of `DgsSpringGraphQLAutoConfiguration`?
module comment: I guess the whole `graphql-dgs-spring-boot-oss-autoconfigure` should be removed, right?

File: graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/GraphqlSSESubscriptionGraphQLClientTest.kt (hunk 5 above)
Comment (paulbakker) [reply graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/GraphqlSSESubscriptionGraphQLClientTest.kt:2]: Weird, I have no idea. The copyright header comes from a profile in Intellij, which has `$today.year` as a variable. After this is merged with can do a clean up.

File: graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/GraphqlSSESubscriptionGraphQLClientTest.kt (hunk 2 above)
Comment (paulbakker) [reply graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/GraphqlSSESubscriptionGraphQLClientTest.kt:48]: Done

File: graphql-dgs-example-shared/build.gradle (hunk 3 above)
Comment (paulbakker) [reply graphql-dgs-example-shared/build.gradle:27]: yes - it's a library that adds file uploads, which is what the Spring folks recommend using as well.

File: graphql-dgs-spring-boot-oss-autoconfigure/src/main/kotlin/com/netflix/graphql/dgs/autoconfig/DgsAutoConfiguration.kt (hunk 7 above)
Comment (paulbakker) [reply graphql-dgs-spring-boot-oss-autoconfigure/src/main/kotlin/com/netflix/graphql/dgs/autoconfig/DgsAutoConfiguration.kt:32]: yes, thanks, I removed it from settings.gradle but forgot to remove the files.

File: graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/WebSocketGraphQLClientWithDGSServerTest.kt [hunk 8]
```diff
@@ -31,25 +31,24 @@ package com.netflix.graphql.dgs.client
...
 import reactor.test.StepVerifier
 
 @SpringBootTest(
-    classes = [DgsWebSocketAutoConfig::class, TestApp::class],
+    classes = [TestApp::class],
     webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
 )
-@EnableAutoConfiguration(exclude = [DgsSSEAutoConfig::class, DgsGraphQLSSEAutoConfig::class])
+@EnableDgsTest
+@Disabled
```
Comment (iparadiso) [inline graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/WebSocketGraphQLClientWithDGSServerTest.kt:51]: Is it intentional to disable this test?

File: settings.gradle.kts [hunk 9]
```diff
@@ -26,29 +26,16 @@ gradleEnterprise {
 }
 rootProject.name = "dgs-framework"
 include("graphql-dgs")
+//include("graphql-dgs-spring-boot-oss-autoconfigure")
```
Comment (iparadiso) [inline settings.gradle.kts:29]: nit:  Clean up dead code

**Review by iparadiso — APPROVED** [review]

**Pull request merged**